spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/mapping/CommonMappingRuleInterpreter.scala

```diff
@@ -27,26 +27,52 @@ import za.co.absa.enceladus.model.MappingTable
 import za.co.absa.enceladus.model.conformanceRule.MappingConformanceRule
 import za.co.absa.enceladus.model.dataFrameFilter.DataFrameFilter
 import za.co.absa.enceladus.conformance.interpreter.rules.ValidationException
+import za.co.absa.enceladus.utils.error.Mapping
+import za.co.absa.enceladus.utils.schema.SchemaUtils
 import za.co.absa.enceladus.utils.validation.ExpressionValidator
+import za.co.absa.spark.hats.transformations.NestedArrayTransformations
 
-import scala.util.Try
+import scala.util.{Failure, Success, Try}
 import scala.util.control.NonFatal
 
 trait CommonMappingRuleInterpreter {
 
   protected val rule: MappingConformanceRule
   protected val log: Logger
 
+  protected def outputColumnNames(): String = rule.allOutputColumns().mkString(", ")
+
+  protected def getOutputsStructColumnName(df: DataFrame): String = SchemaUtils.getClosestUniqueName("outputs", df.schema)
+
+  protected val mappings: Seq[Mapping] = rule.attributeMappings.map {
+    case (mappingTableField, dataframeField) => Mapping(mappingTableField, dataframeField)
+  }.toSeq
```

Comment:
What do you mean?

Reply:
I guess, the question is, why is this a `val`. It can be easilly be `def`, actually considering it's used only at one place can as well be a local `val`.